File: 328/2C.py
#!/usr/bin/python

from fractions import gcd
import sys

f = sys.stdin
inp = [int(x) for x in f.readline().split(' ')]
if inp[1] > inp[2]:
    inp[1], inp[2] = inp[2], inp[1]
lcm = inp[1] / gcd(inp[1], inp[2]) * inp[2]
ans = inp[1] * (inp[0] /  lcm) + min(inp[0] % lcm,  inp[1] - 1)
gc = gcd(ans, inp[0])
print "%d/%d" % (ans / gc, inp[0] / gc)

File: 328/2C.pl
#!/usr/bin/perl
# WA for large numbers;

use bignum;

sub gcd {
    my ($a, $b) = @_;
    ($a, $b) = ($b, $a) if $a > $b;
    while ($a) { ($a, $b) = ($b % $a, $a) }
    return $b;
}

sub min {
    my ($a, $b) = @_;
    ($a, $b) = ($b, $a) if $a > $b;
    return $a;
}

@in = split ' ', <>;
($in[1], $in[2]) = ($in[2], $in[1]) if $in[1] > $in[2];
$lcm = $in[1] / gcd($in[1], $in[2]) * $in[2];
$ans = $in[1] * int($in[0] /  $lcm) + min($in[0] % $lcm,  $in[1] - 1);
$gc = gcd($ans, $in[0]);
print $ans / $gc, '/', $in[0] / $gc, "\n";
